staging/src/k8s.io/apiextensions-apiserver/test/integration/ratcheting_test.go

```diff
@@ -0,0 +1,1345 @@
+	Type: "integer",
+}
+
+var numbersMapSchema *apiextensionsv1.JSONSchemaProps = &apiextensionsv1.JSONSchemaProps{
+	Type: "object",
+	AdditionalProperties: &apiextensionsv1.JSONSchemaPropsOrBool{
+		Schema: numberSchema,
+	},
+}
+
+type RatchetingTestContext struct {
+	*testing.T
+	DynamicClient       dynamic.Interface
+	APIExtensionsClient clientset.Interface
+}
+
+type RatchetingTestOperation = func(ctx *RatchetingTestContext) error
+
+func ExpectError(op RatchetingTestOperation) RatchetingTestOperation {
+	return func(ctx *RatchetingTestContext) error {
+		err := op(ctx)
+		if err != nil {
+			return nil
+		}
+		return errors.New("expected error")
+	}
+}
+
+// apiextensions-apiserver has discovery disabled, so hardcode this mapping
+var fakeRESTMapper map[schema.GroupVersionResource]string = map[schema.GroupVersionResource]string{
+	noxuV1Beta1: "WishIHadChosenNoxu",
+}
+
+// Applies a patch to given GVR and waits for it to be reflected by a GET
+func ApplyPatchOperation(gvr schema.GroupVersionResource, name string, patch map[string]interface{}) RatchetingTestOperation {
+	return func(ctx *RatchetingTestContext) error {
+		// Lookup GVK from discovery
+		kind, ok := fakeRESTMapper[gvr]
+		if !ok {
+			return fmt.Errorf("no mapping found for Gvr %v, add entry to fakeRESTMapper", gvr)
+		}
+
+		patch["kind"] = kind
+		patch["apiVersion"] = gvr.GroupVersion().String()
+
+		if meta, ok := patch["metadata"]; ok {
+			mObj := meta.(map[string]interface{})
+			mObj["name"] = name
+			mObj["namespace"] = "default"
+		} else {
+			patch["metadata"] = map[string]interface{}{
+				"name":      name,
+				"namespace": "default",
+			}
+		}
+
+		_, err := ctx.DynamicClient.Resource(gvr).Namespace("default").Apply(context.TODO(), name, &unstructured.Unstructured{
+			Object: patch,
+		}, metav1.ApplyOptions{
+			FieldManager: "manager",
+		})
+
+		return err
+	}
+}
+
+// Replaces schema used for v1beta1 of noxu crd
+func UpdateNoxuV1Beta1Schema(newSchema *apiextensionsv1.JSONSchemaProps) RatchetingTestOperation {
+	return func(ctx *RatchetingTestContext) error {
+		var noxuDefinition *apiextensionsv1.CustomResourceDefinition
+		var err error = apierrors.NewConflict(schema.GroupResource{}, "", nil)
+		for apierrors.IsConflict(err) {
+			noxuDefinition, err = ctx.APIExtensionsClient.ApiextensionsV1().CustomResourceDefinitions().Get(context.TODO(), "noxus.mygroup.example.com", metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+
+			for _, v := range noxuDefinition.Spec.Versions {
+				if v.Name != "v1beta1" {
+					continue
+				}
+				v.Schema.OpenAPIV3Schema = newSchema
+			}
+
+			_, err = ctx.APIExtensionsClient.ApiextensionsV1().CustomResourceDefinitions().Update(context.TODO(), noxuDefinition, metav1.UpdateOptions{
+				FieldManager: "manager",
+			})
+		}
+		return err
+	}
+}
+
+// Assumes there is only one version of noxu
+func PatchNoxuV1Beta1Schema(patch map[string]interface{}) RatchetingTestOperation {
```

Comment:
please get rid of "noxu" in these tests. We should not cargo cult that even more. Use your own CRD please.

Reply:
the tests only stole the word "noxu" but already had their own CRD. I've renamed it to be avoid confusion